### PR TITLE
CORE-63: add an apps_htcondor_extra table

### PR DESCRIPTION
**Pull request by ianmcorvidae**
This will store extra things to be added to the job requirements at the
app level. This should only be editable by administrators, and will be
joined to the requirements we manage manually by way of an &&.

I pretty much just copied another conversion, so hopefully I didn't do something wrong there, heh.

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
The other approach I could take here, and I'm tempted by it, is to shove this in a side table instead. Which might provide a nicer means of not polluting app listings with this? Dunno?

File: src/main/conversions/v2.25.0/c2_25_0_2019020601.clj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+(ns facepalm.c2-25-0-2019020601
+  (:use [kameleon.sql-reader :only [exec-sql-statement]]))
+
+(def ^:private version
+  "The destination database version"
+  "2.25.0:20190206.01")
```
Comment (psarando) [inline src/main/conversions/v2.25.0/c2_25_0_2019020601.clj:6]: We also have to add this verison to the `src/main/data/99_version.sql` file, for new databases.
```sql
INSERT INTO version (version) VALUES ('2.25.0:20190206.01');
```

**Review by psarando — CHANGES_REQUESTED** [review]
LGTM. We just need a new entry in the `versions` data file, for freshly initialized dbs.

**Comment by psarando** [comment]
> The other approach I could take here, and I'm tempted by it, is to shove this in a side table instead. Which might provide a nicer means of not polluting app listings with this?

That sounds like a fine idea to me as well. It could also set us up with a pattern for expanding additional submission options for other job systems in the future (maybe they each get their own `_extra_requirements` tables? 🤷‍♂️).

**Comment by ianmcorvidae** [comment]
Heh, that's a good point. I think I'll switch it to being a separate, dedicated table. It's not perfectly normalized, but... eh

**Review by psarando — APPROVED** [review]

**Comment by ianmcorvidae** [comment]
I also see that now there's another conversion (dropping collaborators table) that got merged in before this. I'm not sure if I should change the date on this conversion so they're in the order they got merged or not.

**Comment by psarando** [comment]
> I also see that now there's another conversion (dropping collaborators table) that got merged in before this. I'm not sure if I should change the date on this conversion so they're in the order they got merged or not.

I don't think it matters either way, since these conversions don't operate on the same tables or data.

**Comment by slr71** [comment]
Hmm, the only reason I can see it making a difference is if the other conversion was applied before this one. The conversion files are applied in lexographical order, if I remember correctly, so this conversion might not be applied if the one to remove the collaborators table was applied in a different facepalm run.

I don't think that's going to happen, though. The collaborators table conversion has only been applied to QA, and we'll be refreshing the database in order to apply these changes anyway.

**Comment by psarando** [comment]
> The collaborators table conversion has only been applied to QA, and we'll be refreshing the database in order to apply these changes anyway.

Good point. It would make a difference if the other conversion was already applied to production.

**Comment by ianmcorvidae** [comment]
I'm going ahead and getting it merged in, then. I haven't done the apps service work, but, well, this is a benefit of it being another table too, heh :)

**Comment by ianmcorvidae** [comment]
Thanks for the reviews!

**Pull request merged**